skypy/power_spectrum/_halofit.py

```diff
@@ -98,18 +99,45 @@ def halofit(wavenumber, redshift, linear_power_spectrum,
 
     Examples
     --------
+
+    This will return the halofit power spectrum for a given wavenumber,
+    redshift and linear power spectrum:
+
     >>> import numpy as np
     >>> from astropy.cosmology import FlatLambdaCDM
-    >>> kvec = np.array([1.00000000e-04, 1.01000000e+01])
-    >>> zvalue = 0.0
-    >>> pvec = np.array([388.6725682632502, 0.21676249605280398])
     >>> cosmo = FlatLambdaCDM(H0=67.04, Om0=0.21479, Ob0=0.04895)
+
+    The Takahashi halofit power spectrum in :math:`Mpc^3` at redshift 0 and at
+    wavenumers :math:`k=0.0001/Mpc` and :math:`k=10.1/Mpc`:
+
+    >>> zvalue = 0.0
+    >>> kvec = np.array([1.00000000e-04, 1.01000000e+01])
+    >>> pvec = np.array([[388.6725682632502, 0.21676249605280398]])
     >>> halofit(kvec, zvalue, pvec, cosmo, _takahashi_parameters)
-    array([388.67064424,   0.72797614])
+    array([[388.67064424,   0.72797614]])
+
+    Create a plot for the Smith halofit model at redshift 0
+    and Planck15 cosmology:
+
+    >>> from skypy.power_spectrum import halofit
+    >>> from astropy.cosmology import Planck15
+    >>> z = 0.0
+    >>> k = np.logspace(-4.0, 0.0, 5)
+    >>> pk = [[705.54997046, 6474.60158058, 37161.00990355, 9657.02613688,
+    ...           114.60445565]]
+    >>> p = np.array(pk)
+    >>> smith = halofit_smith(k, z, p, Planck15)
+    >>> import matplotlib.pyplot as plt  # doctest: +SKIP
+    >>> plt.loglog(k, smith.T, label='Smith')  # doctest: +SKIP
+    >>> plt.loglog(k,p.T, label='Linear')  # doctest: +SKIP
+    >>> plt.xlabel(r'k $(1/Mpc)$')  # doctest: +SKIP
+    >>> plt.ylabel(r'P $(Mpc^3)$')  # doctest: +SKIP
+    >>> plt.legend()  # doctest: +SKIP
+    >>> plt.show()  # doctest: +SKIP
+
     '''
 
     # Manage shapes of input arrays
-    return_shape = np.shape(linear_power_spectrum)
```

Comment:
I think it might be good to keep this because one would not expect the shape to change between input and output power spectrum, and the `linear_power_spectrum` is necessarily made 2d in the next step for the computations to work.